includes/payments/functions.php

```diff
@@ -161,7 +161,8 @@ function edd_insert_payment( $payment_data = array() ) {
 			'currency'     => $payment_data['currency'],
 			'downloads'    => $payment_data['downloads'],
 			'user_info'    => $payment_data['user_info'],
-			'cart_details' => $payment_data['cart_details']
+			'cart_details' => $payment_data['cart_details'],
+			'tax'          => $cart_tax,
```

Comment:
Good point. I've updated the PR so to update the backwards compat function. It now only copies tax in `_edd_payment_meta` to `_edd_payment_tax` if tax exists (rather than defaulting it to 0). I think that makes more sense than the original patch.